src/core_modules/capture-core/components/WidgetStagesAndEvents/Stages/Stage/StageDetail/StageDetail.component.js

```diff
@@ -0,0 +1,125 @@
+// @flow
+import React, { type ComponentType } from 'react';
+import i18n from '@dhis2/d2-i18n';
+import { withStyles } from '@material-ui/core';
+import {
+    Table,
+    Row,
+    Cell,
+    HeaderCell,
+    Head,
+    Body,
+} from 'capture-ui';
+import { colors, spacersNum } from '@dhis2/ui';
+import { formatValueForView, getValueByKeyFromEvent } from '../helpers';
+import { dataElementTypes } from '../../../../../metaData';
+import type { Props } from './stageDetail.types';
+
+
+const styles = {
+    table: {
+        display: 'block',
+        overflowX: 'auto',
+    },
+    row: {
+        maxWidth: '100%',
+        whiteSpace: 'nowrap',
+    },
+    container: {
+        display: 'flex',
+        padding: spacersNum.dp8,
+        backgroundColor: colors.grey200,
+        alignItems: 'center',
+    },
+};
+const StageDetailPlain = ({ events, data, classes }: Props) => {
+    const defaultColumns = [
```

Comment:
Hey,
I feel that `StageDetailPlain` component has too much logic inside. What do you think about extracting some of this logic outside of the component? I like how you use the` Stages/Stage/helpers.js`, maybe you can do something similar here.
Thanks!

Reply:
Hey, as mentioned above, this is a draft and I am still waiting for the data-table, once it's merged the logic will be extract